pkg/util/grpc/stats/grpc_stats.go

```diff
@@ -0,0 +1,109 @@
+package stats
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc/stats"
+)
+
+func NewStatsHandler(r prometheus.Registerer) stats.Handler {
+	const MiB = 1024 * 1024
+	messageSizeBuckets := []float64{1 * MiB, 2.5 * MiB, 5 * MiB, 10 * MiB, 25 * MiB, 50 * MiB, 100 * MiB, 250 * MiB}
+
+	return &grpcStatsHandler{
+		connectedClients: promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_grpc_connected_clients",
```

Comment:
**Suggestion(non-blocking)**: This struct has a large potential for reuse in other projects. Would you mind making the `cortex` prefix a configurable namespace?

Reply:
Thanks for suggestion. I plan to drop this PR if/when https://github.com/weaveworks/common/pull/196 gets merged. If #196 doesn't get accepted, I will reconsider this feedback.